src/errors/publicErrors.ts

```diff
@@ -0,0 +1,55 @@
+import { constants as httpConstants } from 'node:http2'
+
+import type { FreeformRecord } from '../common/commonTypes'
+
+import { PublicNonRecoverableError } from './PublicNonRecoverableError'
+
+export type CommonErrorParams = {
+  message: string
+  details?: FreeformRecord
+}
+
+export type OptionalMessageErrorParams = {
+  message?: string
+  details?: FreeformRecord
+}
+
+export type ValidationError = {
+  message: string
+  path: string[]
+}
+
```

Comment:
🟢 Could you please also add 
```
export class AccessDeniedError extends PublicNonRecoverableError {
	constructor(params: CommonErrorParams) {
		super({
			message: params.message,
			errorCode: 'ACCESS_DENIED',
			httpStatusCode: 403,
			details: params.details,
		})
	}
}
```
🙏 

We have It on the translation store and would be nice to be able to use it from here

Reply:
done